cloudformation/ec2.yaml

```diff
@@ -0,0 +1,335 @@
+                            "timezone": "UTC"
+                          },
+                          {
+                            "file_path": "/var/log/torchserve/model_metrics.log",
+                            "log_group_name": "${AWS::StackName}/{instance_id}/TorchServe",
+                            "log_stream_name": "TorchServeModelMetrics",
+                            "timezone": "UTC"
+                          }
+                        ]
+                      }
+                    }
+                  }
+                }
+          commands:
+            01_stop_service:
+              command: /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a stop
+            02_start_service:
+              command: /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a fetch-config -m ec2 -c file:/opt/aws/amazon-cloudwatch-agent/etc/amazon-cloudwatch-agent.json -s
+        CFNSetup:
+          files:
+            /etc/cfn/cfn-hup.conf:
+               content: !Sub |
+                 [main]
+                 stack=${AWS::StackId}
+                 region=${AWS::Region}
+                 interval=1
+               mode: '000400'
+               owner: root
+               group: root
+            /etc/cfn/hooks.d/cfn-auto-reloader.conf:
+               content: !Sub |
+                 [cfn-auto-reloader-hook]
+                 triggers=post.update
+                 path=Resources.TorchServeInstance.Metadata.AWS::CloudFormation::Init
+                 action=/opt/aws/bin/cfn-init -v --stack ${AWS::StackId} --resource TorchServeInstance --region ${AWS::Region} --configsets InstallAndRun
+                 runas=root
+               mode: '000400'
+               owner: root
+               group: root
+            /lib/systemd/system/cfn-hup.service:
+                content: !Sub |
+                  [Unit]
+                  Description=cfn-hup daemon
+                  [Service]
+                  Type=simple
+                  ExecStart=/opt/aws/bin/cfn-hup
+                  Restart=always
+                  [Install]
+                  WantedBy=multi-user.target
+          commands:
+            01_enable_cfn_hup:
+              command: !Sub |
+                systemctl enable cfn-hup.service
+            02_start_cfn_hup:
+              command: !Sub |
+                systemctl start cfn-hup.service
+    Properties:
+      ImageId: !FindInMap
+        - AWSRegion2AMI
+        - !Ref 'AWS::Region'
+        - AMI
+      InstanceType: !Ref 'InstanceType'
+      IamInstanceProfile: !Ref CloudwatchInstanceProfile
+      SecurityGroupIds:
+        - !Ref 'TorcheServeSecurityGroup'
+      SubnetId: !Ref 'SubnetId'
+      KeyName: !Ref 'KeyName'
+      UserData:
+        Fn::Base64: !Sub |
+          #!/bin/bash
+          wget https://s3.amazonaws.com/amazoncloudwatch-agent/ubuntu/amd64/latest/amazon-cloudwatch-agent.deb -O /tmp/amazon-cloudwatch-agent.deb
```

Comment:
@maaquib It will be better to use a common install script and call that from the CFN vs yet another place to have the install steps and maintain them.  

This can be done as part optimization for next release.

Reply:
Makes sense. Will pull in the common install scripts once those PRs have bene merged